glean-core/src/lib.rs

```diff
@@ -485,6 +485,7 @@ impl Glean {
     /// * `uuid` - The UUID of the ping in question.
     /// * `status` - The upload result.
     pub fn process_ping_upload_response(&self, uuid: &str, status: UploadResult) {
+        status.record_error(self);
```

Comment:
Note-worthy (before I give this a full look): The Python bindings won't initialize a full Glean anymore in the upload process.
Therefore Python will be unable to report any of those errors
(As this code here is in `src/lib.rs` only, it will all continue to work and not crash or anything)

Reply:
Yes I saw that and it is the reason why I put it in the Glean object. We don't want to record anything in case there is no Glean object, correct?

Reply:
Good catch, @badboy.  We could create a bug for that and not block this -- I still wonder if a good way around that would be to have a pure-Rust uploading binary (which would actually do a full init so it could record these things).

Reply:
Let's file a bug to add support for upload error recording to Python.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1644359